### PR TITLE
docs(notebooks): add LO guide

### PR DESCRIPTION
**Pull request by ritajitmajumdar1**
### Summary

This notebook shows an MVP for using gate cutting to improve the hardware connectivity. 

### Details and comments

- This notebook uses a periodic MBL circuit for demonstration.
- The results from this notebook does not use any error mitigation techniques.

### CODE REVIEW

File: docs/notebooks/circuits.py
Comment (pedrorrivero) [inline docs/notebooks/circuits.py:1]: We cannot have a separate python file to accompany the notebook. 

Please copy-paste the needed contents of the file in the notebook. We also have updated circuit building functionality in the repo if you want to copy-paste that instead.

**Review by pedrorrivero — CHANGES_REQUESTED** [review]
Haven't reviewed the notebook but this needs to be addressed first, marking the PR as draft again until it's resolved so that others don't start reviewing.

File: docs/notebooks/circuits.py
Comment (ritajitmajumdar1) [reply docs/notebooks/circuits.py:1]: Added circuit construction within the ipynb. This notebook requires the periodic version of MBL file. Therefore, keeping the circuit construction local, may be imported from quantum-enablement repo in future.

**Comment by ritajitmajumdar1** [comment]
This notebook currently uses Primitives V1. This is because as per my last discussion with Jim Garrison, circuit-knitting-toolbox can support Primitives V2, but it should be separately installed from source for now. Support for Primitives V2 is not yet available for pip installation. So, I am sticking to Primitives V1 till the time support for V2 is available in pip installation.

**Review by pedrorrivero — CHANGES_REQUESTED** [review]
Points to address:
- [x] We need to add CKT as a notebook dependency to the repo in `pyproject.toml`
- [ ] Please update the circuit generation functions to use:
  - [ ] `compute_uncompute` function (copied and pasted) from our library
  - [ ] No need for a random spin chain, just do `010101...`
  - [ ] Produce the MBL parameters in the same way as we do in the latest version of the scaling notebook
- [ ] Delete the commented `initial_layout`
- [ ] Instead of hardcoding the initial layout manually, I would fix the number of qubits and find a way to generate the initial layout for the circuit (before the final transpilation). And explain why we need to know the initial layout in words.
- [ ] You mention in text that the circuit is 127 qubits, this is not correct
- [ ] Use `Batch` instead of session, since you say that you _execute the circuit in batches_

Other than this I really like the notebook, and I think that including the last section to compare against the uncut circuit is a really nice touch 🙂 

Great job!!

**Comment by ritajitmajumdar1** [comment]
Addressed the comments by @pedrorrivero. For this example, running in Batch is not necessary. So I have removed it to avoid unnecessary complications. We may include it when we scale this up to more depth.

I have not used transpilation to find initial layout. I have mentioned in the notes that it is possible to find the layout via transpilation, but here we do it manually. The reason beings, we have already performed one transpilation with virtual coupling map. Finding good initial layout automatically will require creating a virtual circuit without the long range entanglement, and then again add those entanglements. The first one (virtual coupling map) is absolutely necessary. I am of the opinion to not add the 2nd one to avoid complications. I feel anyone going through the notebook may get lost in these, and miss the primary message.

**Review by pedrorrivero — APPROVED** [review]
Great job @ritajitmajumdar1! LGTM 🥳

**Pull request merged**